### ISSUE TITLE
Hyperlink Lable Woes

### ISSUE THREAD

**Issue by NoFoolLikeOne** [aarronc/hutton-helper#72]
I'm getting some issues where the code goes into a look because resizable objects like the news label can occasionally fight each other. 

I have stopped this on the Canonn plugin by introducing some code to prevent races. I think it would be helpful to add this to the Hutton Helper too as I occasionally get this problem when running the helper.

```
class NewsLink(HyperlinkLabel):

    def __init__(self, parent):

        HyperlinkLabel.__init__(
            self,
            parent,
            text="Fetching News...",
            url=DEFAULT_NEWS_URL,
            wraplength=50,  # updated in __configure_event below
            anchor=tk.NW
        )
        self.resized=False
        self.bind('<Configure>', self.__configure_event)

    def __reset(self):
        self.resized=False;
        
    def __configure_event(self, event):
        "Handle resizing."

        if not self.resized:
            self.resized=True
            self.configure(wraplength=event.width)    
            self.after(500,self.__reset)
```

**Comment by PsymonsB** [comment]
Will look into it tonight. But seems simple enough to implement your code to the Hutton news are any other elements of the hh display causing problems or is it just the news link ?

**Comment by NoFoolLikeOne** [comment]
I think anything that with wrapping potentially can be problematic.

**Comment by NoFoolLikeOne** [comment]
Looks easier said than done. The code above was from my plugin but for whatever reason it doesn't work on HH because it appears to generate a configure event to make the column narrow before doing another to size it properly. With the wait, only the first call is executed so the screen is never sized correctly

I got it to work in HH by forcing a minimum size but the code smells funny to me. I'm sure I can do better

```

class SelfWrappingHyperlinkLabel(HyperlinkLabel):
    "Tries to adjust its width."

    resized=False

    def __init__(self, *a, **kw):
        "Init."

        HyperlinkLabel.__init__(self, *a, **kw)
        self.configure(wraplength=198)
        self.bind('<Configure>', self.__configure_event)



    @classmethod
    def __reset(cls):
        cls.resized=False

    def __configure_event(self, event):
        "Handle resizing."

        if not SelfWrappingHyperlinkLabel.resized:
            newsize=event.width - 2
            if newsize < 198:
                newsize=198
            print("HH reszie {} {}".format(event.width,newsize))
            self.configure(wraplength=newsize)
            SelfWrappingHyperlinkLabel.resized=True
            self.after(500, SelfWrappingHyperlinkLabel.__reset)
```

**Comment by PsymonsB** [comment]
I know this is an old issue but it was never fully resolved. what other plugins do you run alongside the helper because I can't seem to recreate the issue if only using the hutton-helper. it may be other plugins adjusting the width which will cause the helper to re-work out it wrap lengths.

**Comment by PsymonsB** [comment]
think I've eventually found the problem that causes this. the helper selfwrappingtext field never had an initial length set to it. the fix will be in v3.0.2 but if its causing a problem use master

**Comment by PsymonsB** [comment]
fix appears to work as intended
